index.bs

```diff
@@ -277,22 +277,18 @@ interface mixin UnidirectionalStreamsTransport {
    run the following steps:
      1. Let |transport| be the `UnidirectionalStreamsTransport` on which
         `createUnidirectionalStream` is invoked.
+     1. Let |session| be |transport|'s [=[[Session]]=].
      1. If |transport|'s {{WebTransport/state}} is `"closed"` or `"failed"`,
         immediately return a new [=rejected=] promise with a newly created
         {{InvalidStateError}} and abort these steps.
      1. Let |p| be a new promise.
      1. Return |p| and continue the remaining steps [=in parallel=].
-     1. [=SendStream/Create=] a {{SendStream}} with |transport| and |p| when all
-        of the following conditions are met:
-         1. The |transport|'s {{WebTransport/state}} is `"connected"`.
-         1. Stream creation flow control is not being violated by exceeding the
-            max stream limit set by the remote endpoint, as specified in
-            [[!QUIC]].
-         1. |p| has not been [=settled=].
-     1. Queue a task to [=reject=] |p| with a newly created {{InvalidStateError}}
-        when all of the following conditions are met:
-         1. The |transport|'s state is `"closed"` or `"failed"`.
```

Comment:
The updated logic still calls [create an outgoing stream](https://w3c.github.io/webtransport/#session-create-an-outgoing-stream) synchronously, which in turn isn't an algorithm so much as a reference to [Section 4.1](https://datatracker.ietf.org/doc/html/draft-ietf-webtrans-http3/#section-4.1).

That section isn't clearly written as an algorithm, but to the extent we want to treat it as one, the section starts with: _"Once established, both endpoints can open unidirectional streams."_ — this suggests this isn't meant to be called until connection has been established. I.e. There is no waiting logic mentioned, nor would it be possible to on block main thread here, so this would have to fail, which is a breaking change.

@yutakahirano A breaking change is out of scope for this PR I think. I suggest opening a new issue if you want the WG to consider such change in the API.

Reply:
Just to confirm, do you want to [create and outgoing stream](https://w3c.github.io/webtransport/#session-create-an-outgoing-stream) after the session is established?

Reply:
I updated the logic - does it look good?